src/WebJobs.Extensions.DurableTask/DurableHttpRequest.cs

```diff
@@ -170,10 +174,57 @@ public override void WriteJson(JsonWriter writer, object value, JsonSerializer s
                 else
                 {
                     // Don't know how to serialize this - use default behavior, forcing TypeNameHandling.Objects to correctly serialize ITokenSource
-                    var currTypeNameSetting = serializer.TypeNameHandling;
-                    serializer.TypeNameHandling = TypeNameHandling.Objects;
-                    serializer.Serialize(writer, value);
-                    serializer.TypeNameHandling = currTypeNameSetting;
+                    this.CreateTokenSourceSerializer(serializer);
+                    tokenSerializer.Serialize(writer, value);
+                }
+            }
+
+            private void CreateTokenSourceSerializer(JsonSerializer serializer)
```

Comment:
small nits:

1. The method could be `static`.
2. In general, instead of instantiating a variable if it doesn't exist in a void method, and then referencing the variable that was instantiated, an approach often used is to use `GetTokenSourceSerializer(JsonSerializer serializer)`. This way you never directly call `tokenSerializer`, you just call your `Get` method, that you know will never be null. You have effectively accomplished that here, but the Get method approach is a bit less error prone in general.

Neither of these nits are dealbreakers for merging for me.